File: typescript/space-age/space-age.ts
interface Planents {
  [key: string]: number
}

const EARTH_YEAR_SECONDS = 31557600

const orbitalPeriods: Planents = {
  mercury: 0.2408467,
  venus: 0.61519726,
  earth: 1.0,
  mars: 1.8808158,
  jupiter: 11.862615,
  saturn: 29.447498,
  uranus: 84.016846,
  neptune: 164.79132
}

export function age(planet: string, seconds: number): number {
  const earthAge = seconds / EARTH_YEAR_SECONDS
  return Number((earthAge / orbitalPeriods[planet]).toFixed(2))
}

File: go/space-age/space_age.go
// Package space provides a simple way to calculate age on different planets.
package space

// Planet is a type alias for string
type Planet string

var planets = map[Planet]float64{
	"Mercury": 31557600 * 0.2408467,
	"Venus":   31557600 * 0.61519726,
	"Earth":   31557600,
	"Mars":    31557600 * 1.8808158,
	"Jupiter": 31557600 * 11.862615,
	"Saturn":  31557600 * 29.447498,
	"Uranus":  31557600 * 84.016846,
	"Neptune": 31557600 * 164.79132,
}

// Age calculates the age on a given planet.
func Age(seconds float64, planet Planet) float64 {
	orbitalPeriod := planets[planet]
	if orbitalPeriod == 0 {
		return -1
	}
	return seconds / planets[planet]
}
